src/server/ProcessList.cpp

```diff
@@ -85,47 +60,39 @@ using namespace Microsoft::Console::Interactivity;
 // - <none>
 void ConsoleProcessList::FreeProcessData(_In_ ConsoleProcessHandle* const pProcessData)
 {
-    FAIL_FAST_IF(!(ServiceLocator::LocateGlobals().getConsoleInformation().IsConsoleLocked()));
+    assert(ServiceLocator::LocateGlobals().getConsoleInformation().IsConsoleLocked());
 
-    // Assert that the item exists in the list. If it doesn't exist, the end/last will be returned.
-    FAIL_FAST_IF(!(_processes.cend() != std::find(_processes.cbegin(), _processes.cend(), pProcessData)));
-
-    _processes.remove(pProcessData);
-
-    delete pProcessData;
+    const auto it = std::ranges::find(_processes, pProcessData);
+    if (it != _processes.end())
+    {
+        _processes.erase(it);
+        delete pProcessData;
+    }
+    else
+    {
+        // The pointer not existing in the process list would be similar to a heap corruption,
+        // as the only code allowed to allocate a `ConsoleProcessHandle` is us, in AllocProcessData().
+        // An assertion here would indicate a double-free or similar.
+        assert(false);
+    }
 }
 
 // Routine Description:
 // - Locates a process handle in this list.
-// - NOTE: Calling FindProcessInList(0) means you want the root process.
 // Arguments:
-// - dwProcessId - ID of the process to search for or ROOT_PROCESS_ID to find the root process.
+// - dwProcessId - ID of the process to search for.
 // Return Value:
 // - Pointer to the process handle information or nullptr if no match was found.
 ConsoleProcessHandle* ConsoleProcessList::FindProcessInList(const DWORD dwProcessId) const

```

Comment:
you've validated that nobody calls `Find(ROOT)` even unintentionally now?

Reply:
Yeah, there's only a single place calling this function with non-ROOT right now... It's `ApiDispatchers::ServerGenerateConsoleCtrlEvent` which calls it with valid process IDs.